src/tests/ProblemSetBlock.test.tsx

```diff
@@ -496,3 +496,43 @@ test('ProblemSetBlock does not call the onProblemAttempt handler on default cont
 
   expect(queueIbPsetProblemAttemptedV1Event).not.toHaveBeenCalled()
 })
+
+test('Problem level retry limits populate', async () => {
+  const psetContent = `
+  <div class="os-raise-ib-pset" data-retry-limit="1" data-schema-version="1.0">
+    <div class="os-raise-ib-pset-problem" data-retry-limit="2" data-problem-type="input" data-solution="42" data-problem-comparator="integer">
+      <div class="os-raise-ib-pset-problem-content">
+        <p>Input problem content</p>
+      </div>
+      <div class='os-raise-ib-pset-attempts-exhausted-response'>
+        <p>The answer is 42 </p>
+      </div>
+    </div>
+    <div class="os-raise-ib-pset-problem" data-retry-limit="3" data-problem-type="dropdown" data-solution="red" data-solution-options='["red", "blue", "green"]'>
+      <div class="os-raise-ib-pset-problem-content">
+        <p>Dropdown problem content</p>
+      </div>
+    </div>
+    <div class="os-raise-ib-pset-problem" data-retry-limit="4" data-problem-type="multiselect" data-solution='["red", "blue"]' data-solution-options='["red", "blue", "green"]'>
```

Comment:
Just for completeness, can we include a problem that isn't overridden / inherits the retry limit from the parent?

Reply:
Added via latest commit.